server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/Decision.java

```diff
@@ -126,6 +126,11 @@ public boolean higherThan(Type other) {
             return false;
         }
 
+        
+        public boolean canPremptivelyReturn() {
```

Comment:
remove this method please. We preemptively return based on different conditions in different places, so this just adds confusion.